kpatch-build/lookup.c

```diff
@@ -499,20 +496,29 @@ static bool lookup_local_symbol(struct lookup_table *table,
 
 		if (sym->bind == STB_LOCAL && !strcmp(sym->name,
 					lookup_sym->name)) {
-			if (result->objname)
+			if (found)
 				ERROR("duplicate local symbol found for %s",
 						lookup_sym->name);
 
 			result->objname		= table->objname;
 			result->addr		= sym->addr;
 			result->size		= sym->size;
-			result->sympos		= sympos;
 			result->global		= false;
 			result->exported	= false;
+			found = true;
 		}
 	}
+	if (!found)
+		return false;
 
-	return !!result->objname;
+	for_each_obj_symbol(i, sym, table) {
```

Comment:
@liu-song-6 : can you add a short comment in the code explaining that the kernel's kallsyms sympos is determined by address order, and that the given table (as provided by "readelf -s") may not be sorted that way.  Hence the need to iterate through the table a second time to figure out the actual (address ordered) sympos.  Thanks!